packages/web/src/components/list/TreeList.js

```diff
@@ -66,6 +66,253 @@ const transformValueIntoLocalState = (valueArray) => {
 	return valueToSet;
 };
 
+function HierarchicalMenuComponent({
```

Comment:
@mohdashraf010897 It is better to use separate files for components. You can follow the one file for one component rule.